lib/LTP/WhiteList.pm

```diff
@@ -124,12 +130,27 @@ sub is_test_disabled {
 
 sub find_whitelist_testsuite {
     my ($suite) = @_;
+    my $issues;
 
-    my $path = get_var('LTP_KNOWN_ISSUES');
+    my $path = get_var('LTP_KNOWN_ISSUES_LOCAL');
     return undef unless defined($path) and -e $path;
 
     my $content = path($path)->slurp;
-    my $issues = Mojo::JSON::decode_json($content);
+    my $suffix = $path =~ s#.*\.([^.]+)#$1#r;
+
+    if (!$suffix) {
+        die "Using '$suffix', but only JSON and YAML are supported";
+    }
+
+    if ($suffix eq 'json') {
+        $issues = Mojo::JSON::decode_json($content);
+    } elsif ($suffix eq 'yaml') {
+        my $yp = YAML::PP->new;
```

Comment:
```suggestion
        my $yp = YAML::PP->new(schema => [qw/ + Merge /]);
```
I think merge keys are a really cool feature in yaml and I would enable them.

Reply:
Correct me if I'm wrong, but YAML Merge Key draft (https://yaml.org/type/merge.html) is not supported on JSON which we also support. Not sure whether `yamllint` transformation which we use for osd would handle that.

Reply:
Hm, I don't know where `yamllint` is used here sorry! I only know, that the group yaml files of osd do support merge keys as well.
And only for illustration, I don't say that it is more readable, it could be used like this:
```yaml
---

.anchors:
    - &tw { product: opensuse:Tumbleweed }

syscalls:
    ustat01: &blub
    - <<: *tw
      retval: ^1$
      message: ustat() is known to fail with EINVAL on Btrfs. Downstream fixes didn't make it to upstream and were removed. bsc#1194208
    ustat02:
      *blub

ima:
    boot_ltp:
    - <<: *tw
      message: Wrong lsm kernel parameter breaks boot. bsc#1194332
```
But in any case, we could add this later if really needed :)

Reply:
I'll test it tomorrow.
@mdoucha WDYT?

Reply:
FYI it'd fail for `yamllint`:
```
$ yamllint ltp_known_issues.yaml 
ltp_known_issues.yaml
  3:12      error    too many spaces inside braces  (braces)
  3:41      error    too many spaces inside braces  (braces)
```
Maybe we need some switch.
Also conversion we use in osd creates a bit different structure (I'd need to test it whether `{"<<":{"product":"opensuse:Tumbleweed"}` would work):
```
$ perl -MJSON -MYAML::XS -E 'local $/; say to_json(Load(<>));' ltp_known_issues.yaml
# current
{"syscalls":{"ustat01":[{"product":"opensuse:Tumbleweed","message":"ustat() is known to fail with EINVAL on Btrfs. Downstream fixes didn't make it to upstream and were removed. bsc#1194208","retval":"^1$"}],"ustat02":[{"retval":"^1$","product":"opensuse:Tumbleweed","message":"ustat() is known to fail with EINVAL on Btrfs. Downstream fixes didn't make it to upstream and were removed. bsc#1194208"}]}} # old
# merge keys
{"syscalls":{"ustat02":[{"<<":{"product":"opensuse:Tumbleweed"},"message":"ustat() is known to fail with EINVAL on Btrfs. Downstream fixes didn't make it to upstream and were removed. bsc#1194208","retval":"^1$"}],"ustat01":[{"retval":"^1$","message":"ustat() is known to fail with EINVAL on Btrfs. Downstream fixes didn't make it to upstream and were removed. bsc#1194208","<<":{"product":"opensuse:Tumbleweed"}}]},".anchors":[{"product":"opensuse:Tumbleweed"}]}
```

Reply:
YAML::XS doesn't implement merge keys. YAML::PP is the only perl module that can do this.

Reply:
yamllint doesn't care about merge keys. it complains about spaces inside braces. it's a linter :)

Reply:
Ah, thx (I was blind). We could switch to `YAML::PP` in in gitlab CI, but I'd postpone it for separate PR.

Reply:
for convenience I use `yamlpp-load-dump` for testing like:
```bash
echo '---
.blub: &blub
  foo: 'bla'

.bla2: &bla2
  foo: 'shubidu'

test2:
  <<: [*bla2, *blub]
' | yamlpp-load-dump --preserve 0 --merge
```

Reply:
If merge keys are supported in the parser package, there's no harm from enabling them. But I'll leave the decision up to you.

Reply:
BTW, if you use `substr()` to detect format like that and some text editor inserts the Unicode byte order mark (0xFEFF) at the beginning of the file, which is completely invisible to the user and still valid, you'll end up with in the error branch of the format check. It's better to use regex and allow whitespace at the beginning.